### DISCUSSION TITLE
Simple template for code-server

### DISCUSSION THREAD

**Discussion by asalimih** [coder/coder#3466] [General]
Hello,
I'm new in this field. so forgive me if I'm misunderstanding.
I wanted to provide a web based IDE for multiple developers on a remote server. so that each user could log in to the server and open his own instance of the IDE. from what I read in the docs coder could provide this solution for me.
I followed the documentation instructions and now I have a workspace running from the following template:
```
terraform {
  required_providers {
    coder = {
      source = "coder/coder"
    }
  }
}

# This example does not provision any resources. Use this
# template as a starting point for writing custom templates
# using any Terraform resource/provider.
#
# See: https://coder.com/docs/coder-oss/latest/templates

data "coder_workspace" "me" {
}

resource "coder_agent" "dev" {
  os             = "linux"
  arch           = "amd64"
  dir            = "/workspace"
  startup_script = <<EOF
curl -fsSL https://code-server.dev/install.sh | sh
code-server --auth none --port 13337
EOF
}

resource "coder_app" "code-server" {
  agent_id          = coder_agent.dev.id
  name              = "VS Code"
  icon              = "${data.coder_workspace.me.access_url}/icons/vscode.svg"
  url               = "http://localhost:13337"
  relative_path     = true
}

resource "coder_app" "vim" {
  agent_id = coder_agent.dev.id
  name     = "Vim"
  icon     = "${data.coder_workspace.me.access_url}/icons/vim.svg"
  command  = "vim"
}
```
However when I open the workspace in the web UI. there is no button for the IDE. it seems that no agent is running. am I doing anything wrong?
Also I tried the docker-code-server template too. in that case there was an agent named main but again no IDE or Terminal button.

**Comment by matifali** [comment]
I suggest you post in [discord](https://discord.gg/coder).

**Comment by mohammedyusha** [comment]
> Hello, I'm new in this field. so forgive me if I'm misunderstanding. I wanted to provide a web based IDE for multiple developers on a remote server. so that each user could log in to the server and open his own instance of the IDE. from what I read in the docs coder could provide this solution for me. I followed the documentation instructions and now I have a workspace running from the following template:
> 
> ```
> terraform {
>   required_providers {
>     coder = {
>       source = "coder/coder"
>     }
>   }
> }
> 
> # This example does not provision any resources. Use this
> # template as a starting point for writing custom templates
> # using any Terraform resource/provider.
> #
> # See: https://coder.com/docs/coder-oss/latest/templates
> 
> data "coder_workspace" "me" {
> }
> 
> resource "coder_agent" "dev" {
>   os             = "linux"
>   arch           = "amd64"
>   dir            = "/workspace"
>   startup_script = <<EOF
> curl -fsSL https://code-server.dev/install.sh | sh
> code-server --auth none --port 13337
> EOF
> }
> 
> resource "coder_app" "code-server" {
>   agent_id          = coder_agent.dev.id
>   name              = "VS Code"
>   icon              = "${data.coder_workspace.me.access_url}/icons/vscode.svg"
>   url               = "http://localhost:13337"
>   relative_path     = true
> }
> 
> resource "coder_app" "vim" {
>   agent_id = coder_agent.dev.id
>   name     = "Vim"
>   icon     = "${data.coder_workspace.me.access_url}/icons/vim.svg"
>   command  = "vim"
> }
> ```
> 
> However when I open the workspace in the web UI. there is no button for the IDE. it seems that no agent is running. am I doing anything wrong? Also I tried the docker-code-server template too. in that case there was an agent named main but again no IDE or Terminal button.

Did you got the solution ? I am also looking for the same.

**Reply by matifali** [reply]
https://github.com/coder/coder/tree/main/examples/templates/docker-code-server

**Comment by matifali** [comment]
Using only `coder` provider will not create a workspace. You need to create a `vm/container` to run your `code-server` on.
If you want to use `docker` then check this template.
https://github.com/coder/coder/tree/main/examples/templates/docker-code-server

**Reply by mohammedyusha** [reply]
Can I run the code-server on AWS EC2? Actually, I deployed coder on AWS EC2.

**Reply by matifali** [reply]
> Can I run the code-server on AWS EC2? Actually, I deployed coder on AWS EC2.

Yes, you should be able to.

Otherwise, there is a an aws template that creates an instance for you and does everything.
https://github.com/coder/coder/tree/main/examples/templates/aws-linux

**Reply by mohammedyusha** [reply]
I tried to create an aws template. While doing that, I got an error.
![image](https://user-images.githubusercontent.com/113102891/195579641-0ccc8997-7098-4ea0-8d0b-b2f24d364b2e.png)
